components/wifi/esp8266-driver/ESP8266Interface.cpp

```diff
@@ -568,6 +568,11 @@ int ESP8266Interface::socket_send(void *handle, const void *data, unsigned size)
         return NSAPI_ERROR_NO_SOCKET;
     }
 
+    if (!size) {
+        // Firmware limitation
+        return socket->proto == NSAPI_TCP ? 0 : NSAPI_ERROR_UNSUPPORTED;
```

Comment:
Can you add check here for the TCP so that it return zero only if the stream is still connected.

So if the OOB handler have already catched `socket:%d:closed` message, we could already return NSAPI_ERROR_CONNECTION_LOST.

Reply:
This is already part of PR #9503 which is under review

Reply:
OK good.